src/compiler/scala/tools/nsc/typechecker/Namers.scala

```diff
@@ -632,14 +643,63 @@ trait Namers extends MethodSynthesis {
         )
       }
 
-      sym setInfo {
-        mkTypeCompleter(copyDef) { sym =>
-          assignParamTypes()
-          lazyType complete sym
-        }
+      mkTypeCompleter(copyDef) { sym =>
+        assignParamTypes()
+        lazyType complete sym
```

Comment:
should we use `CompleterWrapper` here as well? maybe we have the same bug caused by a missing `typeParams` override lingering around.